tests/dist/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+file(GLOB_RECURSE TEST_FILES ${CMAKE_CURRENT_LIST_DIR} test_*.cpp)
```

Comment:
Could we avoid the globbing here? I know it's in our To-Do list, but we might as well enforce it already.

Reply:
I agree with removing globbing in the main source, but in the tests we won't know if we've missed a file (e.g. if I add a new test to the source tree and forget to put it in the `CMakeLists.txt` it won't run), so I actually would argue for keeping the globbing in the tests. The only downside to doing it is that we sometimes have to rerun `cmake` to pick up new tests right? I personally don't find that too much of a hassle given the benefits.

Reply:
:+1: